src/vm/active_plan.rs

```diff
@@ -52,22 +32,8 @@ pub trait ActivePlan<VM: VMBinding> {
     /// The caller needs to make sure that the thread is a mutator thread.
     fn mutator(tls: VMMutatorThread) -> &'static mut Mutator<VM>;
 
-    /// Reset the mutator iterator so that `get_next_mutator()` returns the first mutator.
-    fn reset_mutator_iterator();
-
-    /// Return the next mutator if there is any. This method assumes that the VM implements stateful type
-    /// to remember which mutator is returned and guarantees to return the next when called again. This does
-    /// not need to be thread safe.
-    fn get_next_mutator() -> Option<&'static mut Mutator<VM>>;
-
     /// A utility method to provide a thread-safe mutator iterator from `reset_mutator_iterator()` and `get_next_mutator()`.
-    fn mutators<'a>() -> SynchronizedMutatorIterator<'a, VM> {
-        SynchronizedMutatorIterator {
-            _guard: Self::global().base().mutator_iterator_lock.lock().unwrap(),
-            start: true,
-            phantom: PhantomData,
-        }
-    }
+    fn mutators<'a>() -> impl Iterator<Item = &'a mut Mutator<VM>>;
```

Comment:
You can try something like this. This would work without nightly Rust. It might be slightly slower, but it doesn't matter in this case.
```suggestion
    fn mutators<'a>() -> Box<dyn Iterator<Item = &'a mut Mutator<VM>>>;
```